### PR TITLE
Tunnelling Improvements

### PR DESCRIPTION
**Pull request by wrboyce**
---
##### Contributor checklist

- [x] Build output has been rebuilt and tested
- [ ] Introduces security considerations
- [ ] Tests are included
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: src/utils/device.ts [hunk 1]
```diff
@@ -22,10 +22,14 @@ import * as memoize from 'memoizee';
 import { balenaApi } from '.';
 import { APIError, captureException } from './errors';
 
+const VPN_GUEST_API_KEY = process.env.VPN_GUEST_API_KEY;
```
Comment (Page-) [inline src/utils/device.ts]:
```suggestion
const { VPN_GUEST_API_KEY } = process.env;
```

File: src/proxy-worker.ts [hunk 2]
```diff
@@ -64,28 +64,38 @@ class Tunnel extends nodeTunnel.Tunnel {
...
+				metrics.inc(Metrics.ActiveTunnels);
+				metrics.inc(Metrics.TotalTunnels);
+				socket.on('close', () => {
+					metrics.dec(Metrics.ActiveTunnels);
+				});
+				return socket;
+			} else {
 				const vpnHost = await device.getDeviceVpnHost(uuid, auth);
 				if (vpnHost.id === this.serviceId) {
+					client.end('HTTP/1.0 500 Internal Server Error\r\n\r\n');
```
Comment (Page-) [inline src/proxy-worker.ts]: It might make sense to break this out to a const since it's repeated a bunch of times

File: src/proxy-worker.ts [hunk 3]
```diff
@@ -64,28 +66,38 @@ class Tunnel extends nodeTunnel.Tunnel {
...
+
+		try {
+			if (await deviceIsLocal()) {
+				this.logger.info(`connecting to ${host}:${port}`);
+				const socket = await super.connect(port, host, client, req);
+				metrics.inc(Metrics.ActiveTunnels);
+				metrics.inc(Metrics.TotalTunnels);
+				socket.on('close', () => {
+					metrics.dec(Metrics.ActiveTunnels);
+				});
```
Comment (Page-) [inline src/proxy-worker.ts]: You could maybe extend this with
```suggestion
				socket.on('close', (hadError) => {
					metrics.dec(Metrics.ActiveTunnels);
					if (hadError) {
						metrics.inc(Metrics.TunnelSocketTransmissionErrors);
					}
				});
```
to get info on the number of sockets that closed due to transmission errors as opposed to ones that closed gracefully

Comment (wrboyce) [reply src/proxy-worker.ts]: Nice idea, I'll add that now!

**Pull request merged**